### PR TITLE
Adding support for Level-1C products

### PR DESCRIPTION
**Pull request by constantinius**

### CODE REVIEW

File: src/stactools/sentinel2/constants.py [hunk 1]
```diff
@@ -80,6 +80,11 @@
                 description='Band 9 - Water vapor',
                 center_wavelength=0.945,
                 full_width_half_max=0.026),
+    'B10':
+    Band.create(name='B10',
+                description='Band 10 - SWIR - Cirrus',
+                center_wavelength=1.3735,
+                full_width_half_max=0.026),  # TODO: find out fwhm
```
Comment (jonas-eberle) [inline src/stactools/sentinel2/constants.py:87]: The full_width_half_max for Band 10 is 0.075 for Sentinel-2A (the wavelength numbers are slightly different for Sentinel-2B!).

**Comment by cuttlefish** [comment]
Looks like the tests are failing due to the added band. Maybe the test data needs to be updated?

File: src/stactools/sentinel2/constants.py (hunk 1 above)
Comment (constantinius) [reply src/stactools/sentinel2/constants.py]: Thanks @jonas-eberle! Worked this in.

**Comment by constantinius** [comment]
The test data is fine. The issue is, that Level-2A is missing a band (B10) by design. I updated the tests to offset this difference.

**Review by gadomski — APPROVED** [review]

**Review by lossyrob — CHANGES_REQUESTED** [review]
Thanks for the PR! This all looks good - can you add some L1C test data files and unit tests to show that these changes allow for proper parsing of those data files?

**Comment by constantinius** [comment]
@lossyrob I added a test product and adapted the tests a bit to reflect Level 1C

**Review by lossyrob — APPROVED** [review]

**Comment by constantinius** [comment]
@lossyrob that last CI error seems to be unrelated to this PR

**Comment by gadomski** [comment]
Yup, that issue was just fixed in https://github.com/stac-utils/stactools/commit/b7a6c42a22a271fef11c8bf1b7619c008ca47e91. If you update your stactools dependency to `stactools==0.2.3` you should pick up the fix.

**Pull request merged**